controllers/factory/builders.go

```diff
@@ -525,3 +523,32 @@ func formatContainerImage(globalRepo string, containerImage string) string {
 	}
 	return globalRepo + containerImage[len("quay.io/"):]
 }
+
+func addStrictSecuritySettingsToPod(p *v1.PodSecurityContext, enableStrictSecurity bool) *v1.PodSecurityContext {
+	if !enableStrictSecurity || p != nil {
+		return p
+	}
+	return &v1.PodSecurityContext{
+		RunAsNonRoot: pointer.Bool(true),
+		// 'nobody' has uid `65534' in alpine image
```

Comment:
It's worth noting that in addition to alpline-based VM images, we also use:

- `quay.io/prometheus-operator/prometheus-config-reloader` based on busybox
- `jimmidyson/configmap-reload` based on debian
- `prom/alertmanager` based on busybox

Fortunately they all use `uid=65534` for nobody as well, so it seems to me that adding new parameters for such a small thing would be unnecessary.

However, we do have the ability to override images in the operator, so it seems to me that:

- It's worth correcting this comment a bit since it's not just about alpine.
- It should be written in the documentation about exactly what enabling the parameter  `EnableStrictSecurity` does. In particular about the use of `uid=65534`. Also we should specify that "if you use alternative images, you will need to either use an image with the same uid, or specify security settings explicitly".

Reply:
>Fortunately they all use uid=65534 for nobody as well

Yeah, '65534' for 'nobody' looks like a reserved one.

Fixed the comments, thanks)